packages/core/strapi/src/node/create-build-context.ts

```diff
@@ -79,15 +80,25 @@ const createBuildContext = async <TOptions extends BaseOptions>({
     });
 
   const serverUrl = strapiInstance.config.get<string>('server.url');
-  const adminPath = strapiInstance.config.get<string>('admin.path');
+  const serverAbsoluteUrl = strapiInstance.config.get<string>('server.absoluteUrl');
+  const adminUrl = strapiInstance.config.get<string>('admin.url');
+  const adminAbsoluteUrl = strapiInstance.config.get<string>('admin.absoluteUrl');
+
+  const sameOrigin = new URL(adminAbsoluteUrl).origin === new URL(serverAbsoluteUrl).origin;
+
+  const adminPublicPath = new URL(adminAbsoluteUrl).pathname;
+
+  const adminPath = sameOrigin
+    ? strings.getCommonPath(serverUrl, adminUrl)
```

Comment:
If both path share the same origin, shouldn't we take what's after the common path in the admin URL? Here it'll return the common origin and/or path (or am I missing something?)